smithy-utils/src/main/java/software/amazon/smithy/utils/IoUtils.java

```diff
@@ -152,6 +155,35 @@ public static String readUtf8Url(URL url) {
         }
     }
 
+    /**
+     * Computes the sha256 digest of a file.
+     *
+     * @param path Path to file to compute hash for.
+     * @return sha256 digest string.
+     * @throws UncheckedIOException if the specified file could not be read.
+     */
+    public static String computeSha256(Path path) {
+        try (InputStream in = Files.newInputStream(path)) {
+            MessageDigest md = MessageDigest.getInstance("SHA-256");
+            try (DigestInputStream din = new DigestInputStream(in, md)) {
+                byte[] buf = new byte[8192];
+                int n;
+                do {
+                    n = din.read(buf);
+                } while (n > 0);
+            }
+            StringBuilder sb = new StringBuilder();
+            for (byte b: md.digest()) {
+                sb.append(String.format("%02x", b));
+            }
```

Comment:
```suggestion
            StringBuilder sb = new StringBuilder();
            for (byte b : md.digest()) {
                int decimal = (int) b & 0xff;
                String hex = Integer.toHexString(decimal);
                if (hex.length() == 1) {
                    sb.append('0');
                }
                sb.append(hex);
            }
```
Minor, performance: format is quite expensive.